docs: EXPOSED-715 Fix broken custom function examples & move to example project (#2383)
JetBrains/Exposed: documentation-website/Writerside/topics/SQL-Functions.md

Exposed provides basic support for classic SQL functions. This topic consists of definitions for those functions, and their 
usage examples. It also explains how to define [custom functions](#custom-functions).

For the function examples below, consider the following table:

<code-block lang="kotlin" src="exposed-sql-functions/src/main/kotlin/org/example/tables/SalesTable.kt"/>

## How to use functions

> If you want to retrieve an SQL function result from a query using `.select()`, you should declare the function as a variable first:

Reviewer: ```suggestion
To retrieve the result of an SQL function from a query using `.select()`, declare the function as a variable first:
```

<code-block lang="kotlin" src="exposed-sql-functions/src/main/kotlin/org/example/examples/StringFuncExamples.kt" include-lines="35-36"/>

> This function could also be aliased, in the same way that a [table or query could be aliased](DSL-Querying-data.topic#alias):

Reviewer: ```suggestion
You can alias this function in the same way you [alias a table or query](DSL-Querying-data.topic#alias):
```

<code-block lang="kotlin" src="exposed-sql-functions/src/main/kotlin/org/example/examples/StringFuncExamples.kt" include-lines="39-40"/>

> Also, functions could be chained and combined as needed. The example below generates SQL that concatenates the string values

Reviewer: ```suggestion
SQL functions can be chained and combined as needed. The example below generates SQL that concatenates the string values
```

> stored in 2 columns, before wrapping the function in `TRIM()` and `LOWER()`:

Reviewer: ```suggestion
stored in two columns, before wrapping the function in `TRIM()` and `LOWER()`:
```

### Character length
The [`.charLength()`](https://jetbrains.github.io/Exposed/api/exposed-core/org.jetbrains.exposed.sql/char-length.html)
function returns the length, measured in characters, or `null` if the String value is null.

<code-block lang="kotlin" src="exposed-sql-functions/src/main/kotlin/org/example/examples/StringFuncExamples.kt" include-lines="64-65"/>

## Aggregate functions
These functions should most likely be used in queries with [`.groupBy()`](DSL-Querying-data.topic#group-by).
### Min/Max/Average
To get the minimum, maximum, and average values, use the 
[`.min()`](https://jetbrains.github.io/Exposed/api/exposed-core/org.jetbrains.exposed.sql/min.html)
[`.max()`](https://jetbrains.github.io/Exposed/api/exposed-core/org.jetbrains.exposed.sql/max.html)
and [`.avg()`](https://jetbrains.github.io/Exposed/api/exposed-core/org.jetbrains.exposed.sql/avg.html) functions
respectively. These functions can be applied to any comparable expression:

<code-block lang="kotlin" src="exposed-sql-functions/src/main/kotlin/org/example/examples/AggregateFuncExamples.kt" include-lines="19-27"/>

### Sum/Count

> You can also use SQL functions like `SUM()` and `COUNT()` directly with a column expression:

Reviewer: ```suggestion
You can use SQL functions like `SUM()` and `COUNT()` directly with a column expression:
```

<code-block lang="kotlin" src="exposed-sql-functions/src/main/kotlin/org/example/examples/AggregateFuncExamples.kt" include-lines="30-37"/>

### Statistics

> Some databases provide aggregate functions specifically for statistics and Exposed provides support for 4 of these:

Reviewer: ```suggestion
Some databases provide aggregate functions specifically for statistics. Exposed provides support for four of these:
```

The [`CustomFunction`](https://jetbrains.github.io/Exposed/api/exposed-core/org.jetbrains.exposed.sql/-custom-function/index.html) class accepts 
a function name as the first argument and the column type that should be used to handle its results as the second.
After that, you can provide any amount of additional parameters separated by a comma:

<code-block lang="kotlin" src="exposed-sql-functions/src/main/kotlin/org/example/examples/CustomFuncExamples.kt" include-lines="36-42"/>

There are also shortcuts for string, long, and datetime functions:
* [`CustomStringFunction`](https://jetbrains.github.io/Exposed/api/exposed-core/org.jetbrains.exposed.sql/-custom-string-function.html)
* [`CustomLongFunction`](https://jetbrains.github.io/Exposed/api/exposed-core/org.jetbrains.exposed.sql/-custom-long-function.html)
* [`CustomDateTimeFunction`](https://jetbrains.github.io/Exposed/api/exposed-jodatime/org.jetbrains.exposed.sql.jodatime/-custom-date-time-function.html)

Using one of these shortcuts, the example above could be simplified to:

<code-block lang="kotlin" src="exposed-sql-functions/src/main/kotlin/org/example/examples/CustomFuncExamples.kt" include-lines="45-47"/>

> As an additional example, the following could be used in H2 to mimic its `DATEADD()` function in order to calculate a date 3 months before the current date.

Reviewer: ```suggestion
As an additional example, the following could be used in H2 to mimic its `DATEADD()` function in order to calculate a date three months before the current one.
```
